### PR TITLE
feat: coverage tracking

### PR DESCRIPTION
**Pull request by EliseZeroTwo**
This PR implements tracking coverage of multiple projects in a centralised repository.

A small overview of the format can be found here: https://github.com/famedly/coverage-reports/blob/main/Description.md

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
+elif [[ -z $MEOWCOV_PR_NUMBER ]]
+then
+    meow-coverage --github-token $MEOWCOV_GITHUB_TOKEN --repo-name $MEOWCOV_REPO_NAME coverage-run --source-prefix $MEOWCOV_SOURCE_PREFIX --commit-id $MEOWCOV_COMMIT_ID --new-lcov-file $MEOWCOV_NEW_LCOV push
+elif [[ -z $MEOWCOV_OLD_LCOV ]]
+then
+    meow-coverage --github-token $MEOWCOV_GITHUB_TOKEN --repo-name $MEOWCOV_REPO_NAME coverage-run --source-prefix $MEOWCOV_SOURCE_PREFIX --commit-id $MEOWCOV_COMMIT_ID --new-lcov-file $MEOWCOV_NEW_LCOV pull-request --pr-number $MEOWCOV_PR_NUMBER
 else
-    meow-coverage --source-prefix $MEOWCOV_SOURCE_PREFIX --repo-name $MEOWCOV_REPO_NAME --commit-id $MEOWCOV_COMMIT_ID --github-token $MEOWCOV_GITHUB_TOKEN --new-lcov-file $MEOWCOV_NEW_LCOV pull-request --pr-number $MEOWCOV_PR_NUMBER --old-lcov-file $MEOWCOV_OLD_LCOV
+    meow-coverage --github-token $MEOWCOV_GITHUB_TOKEN --repo-name $MEOWCOV_REPO_NAME coverage-run --source-prefix $MEOWCOV_SOURCE_PREFIX --commit-id $MEOWCOV_COMMIT_ID --new-lcov-file $MEOWCOV_NEW_LCOV pull-request --pr-number $MEOWCOV_PR_NUMBER --old-lcov-file $MEOWCOV_OLD_LCOV
```
Comment (chgrv) [inline entrypoint.sh]: Could you add some brief comments to this section explaining the logic? Just something short about the different cases and what the intended result is.

File: src/coverage/html.rs [hunk 2]
```diff
@@ -39,12 +41,32 @@ pub fn make_commit_link(
...
+			Some((start_line, Some(end_line))) =>
+				Cow::Owned(format!("R{}-R{}", start_line, end_line)),
+			Some((line, None)) => Cow::Owned(format!("R{}", line)),
 			None => Cow::Borrowed(""),
 		}
 	)
 }
 
+/// Gather lines that are next to each other
+fn gather_lines(lines: &[u32]) -> Vec<(u32, u32)> {
```
Comment (chgrv) [inline src/coverage/html.rs:55]: A unit test of this function would be good, if nothing else for providing an example of input and expected output for ease of readability.

**Review by emgrav — CHANGES_REQUESTED** [review]
Looks great, two small notes though.

Looking at the whole PR, what are your thoughts on testing meow-coverage itself?

**Comment by EliseZeroTwo** [comment]
> Looking at the whole PR, what are your thoughts on testing meow-coverage itself?

Some of the modules (such as `tracking/models.rs`) could be tested, but a lot of the code is dependent on the GitHub API which is also where most of the issues could come up (They don't specify their APIs well leading to having to assume some undefined behaviors about them, and they have also broke things without documenting them several times that I ran across). For general testing of this repo though I will open an issue because there certainly are places where tests can be added and would be useful.

**Review by emgrav — APPROVED** [review]
Looks good, great job!

As previously discussed more tests would be good, but since there's a plan to add those I think this is ready.

**Pull request merged**